pkg/operator/ceph/cluster/osd/envs.go

```diff
@@ -169,6 +169,7 @@ func osdActivateEnvVar() []v1.EnvVar {
 					Name: "rook-ceph-config"},
 					Key: "mon_host"}}},
 		{Name: "CEPH_ARGS", Value: "-m $(ROOK_CEPH_MON_HOST)"},
+		{Name: "PAUSE_OSD", Value: "false"},
```

Comment:
By default we are not pausing this container.

Reply:
Yes but you can handle the pause just by adding the env var, you don't need to set any value to it, we can just check its existence.

Reply:
It's nice to include the placeholder so that it's easier to modify the pod spec. Otherwise, someone might format the setting incorrectly.

Reply:
Fine, however, this does not solve the bigger problem that this only works for OSD on PVC. So I don't think this is the right approach.

Reply:
@leseb can you suggest another approach which we should go ahead with.

Reply:
True, `getActivateOSDInitContainer()` is only called for PVCs (search for `doActivateOSDInit`). The alternative is to create a new init container that would apply to all scenarios.

But instead of changing the spec permanently like this, I wonder if we can just document how to pause the OSD. For example, can we just document to change the `ceph-osd` command in the main container to something like `bash -c sleep infinity`? Then when they are done they can restore the `ceph-osd` command. Doing this purely with documentation seems good so we can keep the main scenario untouched.

Reply:
So should i go about documenting this procedure , rather than codiyfing this ?

Reply:
We can keep this implementation since the current code change is simple and doesn't affect the main container. Then we could add documentation that tells how to do it this way with PVCs, or with a manual workaround if not using PVCs. But let's see how the documentation looks to see if it's too complicated after you test it.

Reply:
What about this?

```
kubectl -n rook-ceph patch deployment rook-ceph-osd-0 -p '{"spec": {"template": {"spec": {"containers": [{"name": "osd", "command": ["sleep", "infinity"], "args": []}]}}}}'
```

Reply:
I know that sometimes we might need to do things like this for debugging, but I am really hesitant to add env vars for debugging into maintained code. I think if we can just patch the deployment per @leseb's comment, that is much better than making code changes.